examples/vello_editor/src/main.rs

```diff
@@ -0,0 +1,222 @@
+        let RenderState::Suspended(cached_window) = &mut self.state else {
+            return;
+        };
+
+        // Get the winit window cached in a previous Suspended event or else create a new window
+        let window = cached_window
+            .take()
+            .unwrap_or_else(|| create_winit_window(event_loop));
+
+        // Create a vello Surface
+        let size = window.inner_size();
+        let surface_future = self.context.create_surface(
+            window.clone(),
+            size.width,
+            size.height,
+            wgpu::PresentMode::AutoVsync,
+        );
+        let surface = pollster::block_on(surface_future).expect("Error creating surface");
+        self.text.update_layout(size.width as _, 1.0);
+
+        // Create a vello Renderer for the surface (using its device id)
+        self.renderers
+            .resize_with(self.context.devices.len(), || None);
+        self.renderers[surface.dev_id]
+            .get_or_insert_with(|| create_vello_renderer(&self.context, &surface));
+
+        // Save the Window and Surface to a state variable
+        self.state = RenderState::Active(ActiveRenderState { window, surface });
+
+        event_loop.set_control_flow(ControlFlow::Poll);
+    }
+
+    fn suspended(&mut self, event_loop: &ActiveEventLoop) {
+        if let RenderState::Active(state) = &self.state {
+            self.state = RenderState::Suspended(Some(state.window.clone()));
+        }
+        event_loop.set_control_flow(ControlFlow::Wait);
+    }
+
+    fn window_event(
+        &mut self,
+        event_loop: &ActiveEventLoop,
+        window_id: winit::window::WindowId,
+        event: WindowEvent,
+    ) {
+        // Ignore the event (return from the function) if
+        //   - we have no render_state
+        //   - OR the window id of the event doesn't match the window id of our render_state
+        //
+        // Else extract a mutable reference to the render state from its containing option for use below
+        let render_state = match &mut self.state {
+            RenderState::Active(state) if state.window.id() == window_id => state,
+            _ => return,
+        };
+
+        self.text.handle_event(&event);
+        render_state.window.request_redraw();
+        render_state
+            .window
+            .set_cursor(winit::window::Cursor::Icon(winit::window::CursorIcon::Text));
+
+        match event {
+            // Exit the event loop when a close is requested (e.g. window's close button is pressed)
+            WindowEvent::CloseRequested => event_loop.exit(),
+
+            // Resize the surface when the window is resized
+            WindowEvent::Resized(size) => {
+                self.context
+                    .resize_surface(&mut render_state.surface, size.width, size.height);
+                render_state.window.request_redraw();
+                self.text.update_layout(size.width as _, 1.0);
+            }
```

Comment:
This should also update the coordinates of the cursor/selection.

Right now the cursor position becomes stale when you resize the window.

Reply:
I added a `refresh` method to both `Cursor` and `Selection` and inserted a call here.